lib/service/webhook.go

```diff
@@ -0,0 +1,53 @@
+package service
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/getAlby/lndhub.go/common"
+	"github.com/getAlby/lndhub.go/db/models"
+)
+
+func (svc *LndhubService) StartWebhookSubscribtion(ctx context.Context) {
+
+	svc.Logger.Infof("Starting webhook subscription with webhook url %s", svc.Config.WebhookUrl)
+	incomingInvoices := make(chan models.Invoice)
+	outgoingInvoices := make(chan models.Invoice)
+	svc.InvoicePubSub.Subscribe(common.InvoiceTypeIncoming, incomingInvoices)
+	svc.InvoicePubSub.Subscribe(common.InvoiceTypeOutgoing, outgoingInvoices)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case incoming := <-incomingInvoices:
+			svc.postToWebhook(incoming)
+		case outgoing := <-outgoingInvoices:
+			svc.postToWebhook(outgoing)
+		}
+	}
+}
+func (svc *LndhubService) postToWebhook(invoice models.Invoice) {
+
+	payload := new(bytes.Buffer)
+	err := json.NewEncoder(payload).Encode(invoice)
+	if err != nil {
+		svc.Logger.Error(err)
+		return
+	}
+
+	resp, err := http.Post(svc.Config.WebhookUrl, "application/json", payload)
+	if err != nil {
+		svc.Logger.Error(err)
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
```

Comment:
There's not a really elegant way of doing this (you would nee to do a literal 200 =< resp.StatusCode < 300) We could mandate that the callback needs to return exactly `200`. Seems not that demanding since the receiving server also has to parse our `Invoice` object. But I could add a `201` as a valid response code, seems like the rest of the 200 range is more exotic.